all.go

```diff
@@ -118,48 +157,86 @@ func runAll(
 
 	scrapeManager := scrape.NewManager(log.With(logger, "component", "scrape-manager"), db)
 
-	s, err := NewSampler(db, reloaders,
+	sampler, err := NewSampler(db, reloaders,
 		SamplerScraper(scrapeManager),
 		SamplerConfig(configFile),
 	)
 	if err != nil {
 		return nil, err
 	}
-	if err := s.Run(context.TODO(), g, reloadCh); err != nil {
+	if err := sampler.Run(context.TODO(), g, reloadCh); err != nil {
 		return nil, err
 	}
 
+	var sym *symbol.Symbolizer
+	if symbolServerURL != "" {
+		level.Debug(logger).Log("msg", "configuring symbol server", "url", symbolServerURL)
+		c := symbol.NewSymbolServerClient(http.DefaultClient, symbolServerURL)
+		sym = symbol.NewSymbolizer(logger, c)
+	}
+
 	w := NewWeb(mux, db, maxMergeBatchSize, queryTimeout,
 		WebLogger(logger),
 		WebRegistry(reg),
 		WebReloaders(reloaders),
 		WebTargets(func(ctx context.Context) conprofapi.TargetRetriever {
 			return scrapeManager
 		}),
+		WebSymbolizer(sym),
+		WebLogOpts(httpLogOpts...),
 	)
 	if err = w.Run(context.TODO(), reloadCh); err != nil {
 		return nil, err
 	}
 
-	// run the grpc writable API
-	p, err = runStorage(
-		comp,
-		g,
-		p,
-		reg,
-		logger,
-		db,
-		srv.grpcBindAddr,
-		srv.grpcGracePeriod,
-		srv.grpcCert,
-		srv.grpcKey,
-		srv.grpcClientCA,
+	grpcProbe := prober.NewGRPC()
+	statusProber := prober.Combine(
+		probe,
+		grpcProbe,
+		prober.NewInstrumentation(comp, logger, extprom.WrapRegistererWithPrefix("conprof_", reg)),
 	)
+	maxBytesPerFrame := 1024 * 1024 * 2 // 2 Mb default, might need to be tuned later on.
+	s := store.NewProfileStore(logger, db, maxBytesPerFrame)
+
+	confContentYaml, err := objStoreConfig.Content()
 	if err != nil {
 		return nil, err
 	}
 
-	p.Ready()
+	var symStore storepb.SymbolStoreServer
+	if len(confContentYaml) > 0 {
+		bkt, err := client.NewBucket(logger, confContentYaml, reg, comp.String())
```

Comment:
Reading this it's not really clear what kind of bucket this is. Maybe a comment could help.